### DISCUSSION TITLE
CSI snapshot + restic backup

### DISCUSSION THREAD

**Discussion by GiuseppeLoRe** [vmware-tanzu/velero#4703] [Community support Q&A] [answered]
Hi,

I have successfully deployed Velero with CSI snapshot enabled (ceph backend) and sending backups to an S3 endpoint.
I have noticed that only metadata are sent to the S3 endpoint, data stays in the ceph backend as snapshots.
This is already something for cases where there is an accidental deletion of a persistentvolume.
What if the entire ceph cluster brakes? I would lose both the persistentvolumes and their snapshot.
Is restic the solution for this scenario?
Can the restic integration of CSI snapshots work together?

Thanks,

Giuseppe

**Comment by GiuseppeLoRe** [comment]
In the doubt, I decided to try. I enabled restic in my values.yaml and tried the first backup.
I am stuck now at the restic repository initialisation, seems due to the S3 endpoint CA cert ...

"""
~ > restic init --repo=s3:https://tsmtest.cscs.ch:9000/abc --password-file=/tmp/pass  --cache-dir=/tmp/restic  --cacert=/Users/loreg/cscs/clusters/tsm/tsmtest_agentcert.crt
Fatal: create repository at s3:https://tsmtest.cscs.ch:9000/abc failed: client.BucketExists: Get "https://tsmtest.cscs.ch:9000/abc/?location=": x509: certificate relies on legacy Common Name field, use SANs instead
"""

The S3 endpoint certificate is good enough for Velero when not using the restic integration.

**Comment by voarsh2** [comment]
I am curious about this too.
I notice... my snapshots just contain references to snapshots inside CEPH! How can I get a nice .tar file with the contents of the volume?

**Comment by shubham-pampattiwar** [comment, marked as answer]
@GiuseppeLoRe @voarsh2 As far as I know Velero CSI Snapshots and Restic are 2 different solutions for backing up of `persistentVolumes`:
1. Volume Snapshots: Velero allows you to take snapshots of persistent volumes as part of your backups. These snapshots are stored locally on the cluster itself, if the cluster breaks then so do the stored snapshots. Please refer the[ implementation choices section of CSI plugin](https://velero.io/docs/v1.8/csi/) and the associated [warnings too.](https://github.com/vmware-tanzu/velero-plugin-for-csi/#warning)
2. Restic: [Restic](https://velero.io/docs/v1.8/restic/) is a different solution offered by Velero to take backups of persistentVolumes. Velero’s Restic integration was added to give you an out-of-the-box solution for backing up and restoring almost any type of Kubernetes volume. This integration is an addition to Velero’s capabilities, not a replacement for existing functionality. If you’re running on AWS, and taking EBS snapshots as part of your regular Velero backups, there’s no need to switch to using Restic. However, if you need a volume snapshot plugin for your storage platform, or if you’re using EFS, AzureFile, NFS, emptyDir, local, or any other volume type that doesn’t have a native snapshot concept, Restic might be for you.

**Reply by GiuseppeLoRe** [reply]
Dear @shubham-pampattiwar, thanks. In case I use both CSI snapshots and restic, will velero+restic backup the snapshot or the live volume?

**Reply by shubham-pampattiwar** [reply]
@GiuseppeLoRe Restic does not take backup of snapshots, it performs a direct file system copy of the volume.

**Reply by GiuseppeLoRe** [reply]
Dear @shubham-pampattiwar, is it possible to use restic for a backup schedule and use CSI snapshots for another one?
Ideally I would like to use snapshots, let's say daily, as they are very light and fast, and weekly perform a velero/restic backup to create a remote copy of my data.
Is this possible within the same velero deployment?

**Reply by shubham-pampattiwar** [reply]
@GiuseppeLoRe Afaik this should be possible with the same velero deployment (assuming you are installing velero with restic as well as the CSI plugin). Also, you may need 2 different backups, one for restic and one for snapshots.
